packages/dds/tree/src/feature-libraries/treeCursorUtils.ts

```diff
@@ -177,6 +178,12 @@ class StackCursor<TNode extends NodeData>
             this.exitField();
             return false;
         }
+
+        // Skip empty fields during iteration
+        if (this.getField().length === 0) {
```

Comment:
Have you benchmarked how much overhead this (and the one below) adds for MapTree and text Cursors? They currently avoid empty keys so they shouldn't need this case: if it slows them down a few percent we might want to remove and require keysFromNode only include empty fields.

Reply:
Seems like it slows down some of the benchmarks significantly. I think putting the burden on keysFromNode makes sense. I'll do that and benchmark the difference.

Before:

```
✔   Direct: 'canada'                                  5,269,672.7                    ±1.50%         900       75            5.45
✔   cursorToJsonObject(JsonCursor): 'canada'         26,397,734.9                    ±5.77%         153       51            5.68
✔   cursorToJsonObject(TextCursor): 'canada'         32,707,442.5                    ±3.86%         108       54            5.59
✔   cursorToJsonObject(MapCursor): 'canada'          31,026,583.5                    ±3.85%         114       57            5.64
✔   jsonableTreeFromCursor(JsonCursor): 'canada'     35,852,681.0                    ±3.81%          96       48            5.61
✔   jsonableTreeFromCursor(TextCursor): 'canada'     44,378,280.8                    ±4.24%          82       41            5.74
✔   jsonableTreeFromCursor(MapCursor): 'canada'      49,556,620.8                    ±6.72%          76       38            5.72
✔   mapTreeFromCursor(JsonCursor): 'canada'          59,641,893.9                   ±10.36%          66       33            5.79
✔   mapTreeFromCursor(TextCursor): 'canada'          59,500,950.2                    ±6.35%          72       36            5.63
✔   mapTreeFromCursor(MapCursor): 'canada'           61,002,783.3                    ±6.84%          80       40            5.68
✔   sum(JsonCursor): 'canada'                         9,392,415.7                    ±0.41%         483       69            5.45
✔   sum(TextCursor): 'canada'                        13,200,300.0                    ±0.50%         330       66            5.51
✔   sum(MapCursor): 'canada'                         15,164,852.1                    ±0.45%         272       68            5.45
✔   sum-map(JsonCursor): 'canada'                    17,616,136.8                    ±0.40%         219       73            5.45
✔   sum-map(TextCursor): 'canada'                    20,780,093.2                    ±0.34%         189       63            5.53
✔   sum-map(MapCursor): 'canada'                     22,080,899.3                    ±0.47%         177       59            5.50
✔   averageTwoValues(JsonCursor): 'canada'            7,711,584.2                    ±0.34%         574       82            5.43
✔   averageTwoValues(TextCursor): 'canada'           11,690,887.6                    ±0.51%         365       73            5.49
✔   averageTwoValues(MapCursor): 'canada'             9,904,137.9                    ±0.83%         444       74            5.49
✔   Direct: 'twitter'                                 6,097,720.3                    ±1.04%         747       83            5.42
✔   cursorToJsonObject(JsonCursor): 'twitter'        25,316,256.7                    ±2.74%         156       52            5.57
✔   cursorToJsonObject(TextCursor): 'twitter'        31,688,712.9                    ±1.15%         112       56            5.63
✔   cursorToJsonObject(MapCursor): 'twitter'         24,380,555.5                    ±1.28%         162       54            5.55
✔   jsonableTreeFromCursor(JsonCursor): 'twitter'    43,340,035.8                    ±6.71%          86       43            5.78
✔   jsonableTreeFromCursor(TextCursor): 'twitter'    46,183,078.0                    ±2.97%          78       39            5.77
✔   jsonableTreeFromCursor(MapCursor): 'twitter'     43,641,635.5                    ±5.33%          84       42            5.75
✔   mapTreeFromCursor(JsonCursor): 'twitter'         45,557,930.7                    ±5.80%          76       38            5.83
✔   mapTreeFromCursor(TextCursor): 'twitter'         53,828,455.8                    ±7.52%          70       35            5.96
✔   mapTreeFromCursor(MapCursor): 'twitter'          51,043,841.4                    ±9.29%          78       39            5.83
✔   sum(JsonCursor): 'twitter'                        8,948,613.3                    ±0.23%         504       72            5.43
✔   sum(TextCursor): 'twitter'                       14,597,055.6                    ±0.79%         284       71            5.49
✔   sum(MapCursor): 'twitter'                         9,374,745.1                    ±0.85%         468       78            5.42
✔   sum-map(JsonCursor): 'twitter'                   15,717,769.6                    ±0.25%         264       66            5.47
✔   sum-map(TextCursor): 'twitter'                   21,742,179.6                    ±0.55%         180       60            5.50
✔   sum-map(MapCursor): 'twitter'                    16,434,961.4                    ±0.68%         252       63            5.46
✔   averageTwoValues(JsonCursor): 'twitter'             132,943.7                    ±0.15%      38,003       89            5.38
✔   averageTwoValues(TextCursor): 'twitter'             190,555.7                    ±0.15%      26,522       89            5.41
✔   averageTwoValues(MapCursor): 'twitter'              134,036.6                    ±0.31%      36,363       93            5.45
```

After:

```
✔   Direct: 'canada'                                  5,370,747.6                    ±0.97%         891       81            5.33
✔   cursorToJsonObject(JsonCursor): 'canada'         26,005,786.5                    ±5.34%         156       52            5.73
✔   cursorToJsonObject(TextCursor): 'canada'         31,461,568.2                    ±4.45%         112       56            5.56
✔   cursorToJsonObject(MapCursor): 'canada'          32,018,202.5                    ±3.53%         110       55            5.62
✔   jsonableTreeFromCursor(JsonCursor): 'canada'     39,568,948.6                    ±6.15%          92       46            5.72
✔   jsonableTreeFromCursor(TextCursor): 'canada'     48,780,693.2                    ±4.79%          74       37            5.76
✔   jsonableTreeFromCursor(MapCursor): 'canada'      51,817,209.9                    ±6.18%          72       36            5.82
✔   mapTreeFromCursor(JsonCursor): 'canada'          57,198,626.8                    ±8.98%          66       33            5.99
✔   mapTreeFromCursor(TextCursor): 'canada'          66,225,592.8                    ±9.92%          43       43            5.70
✔   mapTreeFromCursor(MapCursor): 'canada'           61,833,861.4                    ±4.39%          42       42            5.66
✔   sum(JsonCursor): 'canada'                        10,387,735.5                    ±0.37%         410       82            5.42
✔   sum(TextCursor): 'canada'                        15,928,974.0                    ±0.42%         260       65            5.46
✔   sum(MapCursor): 'canada'                         16,586,827.3                    ±0.49%         252       63            5.56
✔   sum-map(JsonCursor): 'canada'                    18,627,366.1                    ±0.30%         207       69            5.44
✔   sum-map(TextCursor): 'canada'                    25,981,752.4                    ±0.33%         132       66            5.46
✔   sum-map(MapCursor): 'canada'                     23,939,566.7                    ±0.41%         165       55            5.54
✔   averageTwoValues(JsonCursor): 'canada'            7,398,649.2                    ±0.21%         602       86            5.46
✔   averageTwoValues(TextCursor): 'canada'           11,079,639.8                    ±0.71%         385       77            5.49
✔   averageTwoValues(MapCursor): 'canada'            10,206,601.5                    ±0.98%         444       74            5.52
✔   Direct: 'twitter'                                 6,018,562.1                    ±0.43%         756       84            5.41
✔   cursorToJsonObject(JsonCursor): 'twitter'        28,686,012.1                    ±1.60%         122       61            5.54
✔   cursorToJsonObject(TextCursor): 'twitter'        36,124,818.1                    ±1.89%          98       49            5.59
✔   cursorToJsonObject(MapCursor): 'twitter'         25,945,477.3                    ±1.25%         153       51            5.52
✔   jsonableTreeFromCursor(JsonCursor): 'twitter'    42,356,132.5                    ±5.18%          84       42            5.69
✔   jsonableTreeFromCursor(TextCursor): 'twitter'    50,940,568.5                    ±2.89%          72       36            5.94
✔   jsonableTreeFromCursor(MapCursor): 'twitter'     43,616,228.6                    ±4.03%          82       41            5.74
✔   mapTreeFromCursor(JsonCursor): 'twitter'         49,788,395.0                    ±6.59%          76       38            5.89
✔   mapTreeFromCursor(TextCursor): 'twitter'         55,116,249.6                    ±6.39%          68       34            5.70
✔   mapTreeFromCursor(MapCursor): 'twitter'          47,718,170.0                    ±7.42%          78       39            5.78
✔   sum(JsonCursor): 'twitter'                       13,582,252.5                    ±1.61%         350       70            5.44
✔   sum(TextCursor): 'twitter'                       19,404,282.9                    ±0.64%         201       67            5.50
✔   sum(MapCursor): 'twitter'                        11,362,900.2                    ±0.81%         375       75            5.42
✔   sum-map(JsonCursor): 'twitter'                   19,975,314.0                    ±0.23%         195       65            5.49
✔   sum-map(TextCursor): 'twitter'                   25,340,131.6                    ±0.65%         156       52            5.48
✔   sum-map(MapCursor): 'twitter'                    17,683,949.9                    ±0.56%         219       73            5.47
✔   averageTwoValues(JsonCursor): 'twitter'             122,824.1                    ±0.17%      40,945       95            5.39
✔   averageTwoValues(TextCursor): 'twitter'             177,810.2                    ±0.33%      28,179       93            5.43
✔   averageTwoValues(MapCursor): 'twitter'              142,460.7                    ±0.18%      35,399       91            5.40
```

Reply:
I think it fine if you want to merge this as is, and do the perf tweaking in a followup PR.

Reply:
Just pushed the change. It's significantly faster but I think you should review the change. I also made a change to a test case so that the round trip tests pass.